lib/yard-settings/handler.rb

```diff
@@ -0,0 +1,48 @@
+# :nodoc:
+class SettingsHandlerBase < YARD::Handlers::Ruby::Base
+  handles method_call :string_setting
+  handles method_call :number_setting
+  handles method_call :boolean_setting
+
+  namespace_only
+
+  def process
+    name = statement.parameters.first.jump(:tstring_content, :ident).source
+    object = YARD::CodeObjects::MethodObject.new(namespace, name)
+    register(object)
```

Comment:
This is the "render each of the settings as a method on the Module" bit. Perhaps this is over-reaching, as the non-ENV-settable settings (which are also available as method calls via method_missing) will not be represented here.

Example of what the current output is:

![screenshot 2016-04-30 15 18 43](https://cloud.githubusercontent.com/assets/211/14936008/1d841e08-0ee7-11e6-8e0e-a14d63eb6f07.png)